lib/libspl/os/freebsd/mnttab.c

```diff
@@ -163,10 +127,53 @@ statfs_init(void)
 	return (error);
 }
 
+#define	BUFSIZE (sizeof (gsfs->f_fstypename) + sizeof (gsfs->f_mntfromname) + \
+    sizeof (gsfs->f_mntonname) + MNTMAXSTR)
+
+static __thread char buf[BUFSIZE];
+
+void
+statfs2mnttab(struct statfs *sfs, struct mnttab *mp)
+{
+	static char mntopts[MNTMAXSTR];
+	long flags;
+
+	mntopts[0] = '\0';
+
+	flags = sfs->f_flags;
+#define	OPTADD(opt)	optadd(mntopts, sizeof (mntopts), (opt))
+	if (flags & MNT_RDONLY)
+		OPTADD(MNTOPT_RO);
+	else
+		OPTADD(MNTOPT_RW);
+	if (flags & MNT_NOSUID)
+		OPTADD(MNTOPT_NOSETUID);
+	else
+		OPTADD(MNTOPT_SETUID);
+	if (flags & MNT_UPDATE)
+		OPTADD(MNTOPT_REMOUNT);
+	if (flags & MNT_NOATIME)
+		OPTADD(MNTOPT_NOATIME);
+	else
+		OPTADD(MNTOPT_ATIME);
+	OPTADD(MNTOPT_NOXATTR);
+	if (flags & MNT_NOEXEC)
+		OPTADD(MNTOPT_NOEXEC);
+	else
+		OPTADD(MNTOPT_EXEC);
+#undef	OPTADD
+	char *p = buf, *e = buf + BUFSIZE;
```

Comment:
Is there some reason I'm not seeing for preferring this mapping to just defining 4 separate __thread variables (or a struct with them, if you like) and referencing into that?

It seems unnecessary.